### ISSUE TITLE
drawImage not always working after 2.0.0-alpha.14+ update

### ISSUE THREAD

**Issue by Koyamie** [Automattic/node-canvas#1260]
## Issue or Feature

After updating canvas to version 2.0.0-alpha.14+ the drawImage function is often not working even with the same code before and after the update, works great in 2.0.0-alpha.13.

## Steps to Reproduce

```js
var Canvas = require('canvas');
const canvas = Canvas.createCanvas(1024, 450);
const ctx = canvas.getContext('2d');
const base = new Image();
base.onload = () => ctx.drawImage(base, 0, 0, 1024, 450);
base.src = Buffer;
```
Buffer is a node Buffer, the base image is not appearing in lot of cases but is always working in version 
2.0.0-alpha.13.

## Your Environment
* Version of node-canvas (e.g. 1.4.0): 2.0.0-alpha.14+
* Environment (e.g. node 4.2.0 on Mac OS X 10.8): node 10.9.0

**Comment by LinusU** [comment]
When are you outputting the image? I think what's happening is that at the point where you save the image, the `base.onload` hasn't fired yet, and thus the image hasn't been drawn at that point...

**Comment by Koyamie** [comment]
Well, this issue seems fixed since 2.0.0-alpha.17.
